File: Codeforces/1478C.cpp
#include<bits/stdc++.h>
 
using namespace std;
 
int main(){
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    //freopen("input.txt","r",stdin);
    int nTest; cin >> nTest;
    while (nTest--){
        int n; cin >> n;
        vector<int64_t> b(2*n);
        vector<int64_t> a;
        map<int64_t, int> mp;
        for(int i=0; i<2*n; i++){
            cin >> b[i];
            mp[b[i]]++;
        }
        bool f = 1;
        sort(b.begin(), b.end());
        reverse(b.begin(), b.end());
        for(int i=0; i<2*n; i++){
            if (mp[b[i]]!=2){
                f = 0;
                break;
            }
        }
        if (!f){
            cout << "NO\n";
            continue;
        }
        for(int i=0; i<2*n; i+=2){
            a.push_back(b[i]);
        }
        int64_t curSub = 0;
        
        mp.clear();
        for(int i=0; i<n; i++){
            if ((a[i] - 2 * curSub) % (2ll * (n - i)) == 0 && (a[i] - 2 * curSub) > 0){
                mp[(a[i] - 2 * curSub) / (2ll * (n - i))]++;
                if (mp[(a[i] - 2 * curSub) / (2ll * (n - i))]>1){
                    f = 0;
                    break;
                }
                curSub += (a[i] - 2 * curSub) / (2ll * (n - i));
            }
            else{
                f = 0;
                break;
            }
        }
        if (f) cout << "YES\n";
        else cout << "NO\n";
    }
    return 0;
}

File: Codeforces/1478C.py
nTest = int(input())

for iTest in range(nTest):
    n = int(input())
    x = input()
    a = []
    d = {}
    for i in x.split():
        a.append(int(i))
        if int(i) not in d:
            d[int(i)] = 1
        else:
            d[int(i)] += 1
    f = 1
    for i in d:
        if d[i] != 2:
            f = 0
            break
    if not f:
        print("NO")
        continue
    a.sort()
    s = set()
    for x in a:
        s.add(x)
    if len(s) != n:
        print("NO")
        continue
    a.clear()
    for x in s:
        a.append(x)
    a.sort()
    a.reverse()
    sum = 0
    
    for i in range(n):
        if a[i] - 2 * sum <= 0 or (a[i] - 2 * sum) % (2 * (n - i)) != 0:
            f = 0
            break
        sum += (a[i] - 2 * sum) / (2 * (n - i))
    if f:
        print("YES")
    else:
        print("NO")